packages/protobuf/src/private/names.ts

```diff
@@ -275,3 +266,43 @@ const reservedMessageProperties: { [k: string]: boolean } = {
   // names reserved by the runtime for the future
   toObject: true,
 };
+
+type Fallback<T extends string> = `${T}$`;
+
+const fallback = <T extends string>(word: T): Fallback<T> => `${word}$`;
+
+type ReservedObjectProperty = keyof typeof reservedObjectProperties;
+
+const isReservedObjectProperty = <T extends string>(
+  word: T | ReservedObjectProperty
+): word is ReservedObjectProperty =>
+  reservedObjectProperties[word as ReservedObjectProperty];
+
+export const safeObjectProperty = <T extends string>(
+  word: T
+): T extends ReservedObjectProperty ? Fallback<T> : T => {
+  if (isReservedObjectProperty(word)) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-explicit-any -- required for narrowing
+    return fallback(word) as any;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-explicit-any -- required for narrowing
+  return word as any;
+};
+
+type ReservedIdentifier = keyof typeof reservedIdent;
+
+const isReservedIdentifier = <T extends string>(
```

Comment:
We just want to conditionally append an `$` and a clear code path is valuable here. Ideally, the naming logic is easy to port to other languages and other code generators, so that they can interact. Can you remove is*()?

Reply:
you're quick! they were removed before I saw this comment e4f43261f355c39a539528e0fa16409b80e8c044